packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,145 @@
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import Web3 from "web3";
+import { parseAncillaryData } from "@uma/common";
+import { BN } from "../types";
+import type { Logger } from "winston";
+import { InsuredBridgeL1Client, Relay } from "../clients/InsuredBridgeL1Client";
+import { InsuredBridgeL2Client, Deposit } from "../clients/InsuredBridgeL2Client";
+
+const { toBN, toWei } = Web3.utils;
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+interface RelayAncillaryData {
+  depositId: number;
+  depositTimestamp: number;
+  recipient: string;
+  l2Sender: string;
+  l1Token: string;
+  amount: number;
+  slowRelayFeePct: number;
+  instantRelayFeePct: number;
+  quoteTimestamp: number;
+  realizedLpFeePct: number;
+  depositContract: string;
+  priceRequestTime: string;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond "Yes" or "No"
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly toBNWei: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+    this.toBNWei = (_number) => toBN(toWei(_number.toString()));
```

Comment:
nit: I dont like this method. Can you define it like [this](https://github.com/UMAprotocol/protocol/blob/8bf6271fd13f66b04ba3036fb2f9f785f076fd4f/packages/trader/src/RangeTrader.ts#L6)?

Reply:
Ah yep, will remove from class.